v2/slogger/logger.go

```diff
@@ -0,0 +1,349 @@
+		pc, file, line, ok := runtime.Caller(skip)
+		if !ok || !containsAnyIgnoredFilename(file) {
+			return pc, file, line, ok
+		}
+	}
+	return 0, "", 0, false
+}
+
+func (self *Logger) logf(level Level, errorCode ErrorCode, messageFmt string, context *Context, args ...interface{}) (*Log, []error) {
+	var errors []error
+
+	for _, filter := range self.TurboFilters {
+		if filter(level, messageFmt, args) == false {
+			return nil, errors
+		}
+	}
+
+	pc, file, line, ok := nonSloggerCaller()
+	if ok == false {
+		return nil, []error{fmt.Errorf("Failed to find the calling method.")}
+	}
+
+	file = stripDirectories(file, self.StripDirs)
+	log := &Log{
+		Prefix:     self.Prefix,
+		Level:      level,
+		ErrorCode:  errorCode,
+		Filename:   file,
+		FuncName:   baseFuncNameForPC(pc),
+		Line:       line,
+		Timestamp:  time.Now(),
+		MessageFmt: messageFmt,
+		Args:       args,
+		Context:    context,
+	}
+
+	for _, appender := range self.Appenders {
+		if err := appender.Append(log); err != nil {
+			error := fmt.Errorf("Error appending. Appender: %T Error: %v", appender, err)
+			errors = append(errors, error)
+		}
+	}
+
+	return log, errors
+}
+
+type Level uint8
+
+// The level is in an order such that the expressions
+// `level < WARN`, `level >= INFO` have intuitive meaning.
+const (
+	OFF Level = iota
+	DEBUG
+	INFO
+	WARN
+	ERROR
+	FATAL
+	topLevel
+)
+
+var strToLevel map[string]Level
+
+var levelToStr []string
+
+func init() {
+	strToLevel = map[string]Level{
+		"off":   OFF,
+		"debug": DEBUG,
+		"info":  INFO,
+		"warn":  WARN,
+		"error": ERROR,
+		"fatal": FATAL,
+	}
+
+	levelToStr = make([]string, len(strToLevel))
+	for str, level := range strToLevel {
+		levelToStr[uint8(level)] = str
+	}
+}
+
+func NewLevel(levelStr string) (Level, error) {
+	level, ok := strToLevel[strings.ToLower(levelStr)]
+
+	if !ok {
+		err := UnknownLevelError{levelStr}
+		return OFF, err
+	}
+
+	return level, nil
+}
+
+func (self Level) String() string {
+	return self.Type()
+}
+
+func (self Level) Type() string {
+	if self >= topLevel {
+		return "off?"
+	}
+
+	return levelToStr[uint8(self)]
+}
+
+type ErrorCode uint8
+
+const NoErrorCode = 0
+
+func stacktrace() []string {
+	ret := make([]string, 0, 2)
+	for skip := 2; true; skip++ {
+		_, file, line, ok := runtime.Caller(skip)
+		if ok == false {
+			break
+		}
+
+		ret = append(ret, fmt.Sprintf("at %s:%d", stripDirectories(file, 2), line))
+	}
+
+	return ret
+}
+
+type StackError struct {
+	Message    string
+	Stacktrace []string
+}
+
+func NewStackError(messageFmt string, args ...interface{}) *StackError {
+	return &StackError{
+		Message:    fmt.Sprintf(messageFmt, args...),
+		Stacktrace: stacktrace(),
+	}
+}
+
+func (self *StackError) Error() string {
+	return fmt.Sprintf("%s\n\t%s", self.Message, strings.Join(self.Stacktrace, "\n\t"))
+}
+
+func stripDirectories(filepath string, toKeep int) string {
+	var idxCutoff int
+	if idxCutoff = strings.LastIndex(filepath, "/"); idxCutoff == -1 {
```

Comment:
[question] I'm probably just missing it, but I would've guessed there'd be a `_windows.go` version of this method for compatibility? Or is it maybe that here windows path (`\`) is not required?